docs/DesignDocs/SwitchlessCalls.md

```diff
@@ -0,0 +1,84 @@
+Context-switchless Calls
+================
+
+## Motivation
+
+Context-switchless Calls are designed to reduce the cost of context switching between hosts and enclaves.
+In an enclave application, the host makes **ECALL**s into functions exposed by the enclaves it created. Likewise,
+the enclaves may make **OCALL**s into functions exposed by the host that created them. In either case, the
+execution has to be transitioned from an untrusted environment to a trusted environment, or vice versa. Since the
+transition is costly due to heavy security checks, it might be more performance advantageous to make the calls
+**context-switchless**: the caller delegates the function call to a worker thread in the other environment, which
+does the real job of calling the function and post the result to the caller. Both the calling thread and the
```

Comment:
grammar: "post" -> "posting"

Reply:
Thank you for the proofreading. @dthaler . All errors are fixed.